Getting type errors in the Response Object

I'm getting a type error in the kubetest.response.Response object after switching from 0.3.2 to 0.3.3. Using the http_proxy_get method from the Pod object I get the following error after sending any request on an endpoint (that works in 0.3.2). 

```
data = resp.json()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <kubetest.response.Response object at 0x000002E181349780>

    def json(self):
        """Convert the response data to JSON.
    
        If the response data is not valid JSON, an error will be raised.
    
        Returns:
            dict: The JSON data.
        """
        # This should generally be the case, since the primary source of Response
        # instances are the HTTP methods on Pods (e.g. http_proxy_get), where
        # `_preload_content=False`. Should that be set to True, it will not return
        # the HTTPResponse but will instead return the response data formatted as
        # the type specified in the `response_type` param.
        #
        # Note that I have found setting _preload_content to True (the default value
        # for the Kubernetes client) to be difficult to deal with from a generic interface
        # because it requires you to know the expected output ("str", "dict", ...)
        # and pass that in as a param, where that could really be abstracted away from
        # the user. For that reason, it is currently set to False -- this may change
        # in the future if desired.
        if isinstance(self.data, urllib3.HTTPResponse):
            return json.loads(self.data.data)
    
        # the response data comes back as a string formatted as a python
        # dictionary might be, where the inner quotes are single quotes
        # (') instead of the double quotes (") expected by JSON standard.
        # to remedy this, we just replace single quotes with double quotes.
>       data = self.data.replace("'", '"')
E       TypeError: a bytes-like object is required, not 'str'

..\kubetest\response.py:59: TypeError
```

--- a/kubetest/__init__.py
+++ b/kubetest/__init__.py
@@ -1,7 +1,7 @@
 """kubetest -- a Kubernetes integration test framework in Python."""
 
 __title__ = 'kubetest'
-__version__ = '0.3.3'
+__version__ = '0.3.4'
 __description__ = 'A Kubernetes integration test framework in Python.'
 __author__ = 'Vapor IO'
 __author_email__ = 'vapor@vapor.io'

--- a/kubetest/objects/pod.py
+++ b/kubetest/objects/pod.py
@@ -224,7 +224,7 @@ class Pod(ApiObject):
                 response_type='str',
                 auth_settings=auth_settings,
                 _return_http_data_only=False,  # we want all info, not just data
-                _preload_content=False,
+                _preload_content=True,
                 _request_timeout=None,
                 collection_formats={}
             ))
@@ -293,7 +293,7 @@ class Pod(ApiObject):
                 response_type='str',
                 auth_settings=auth_settings,
                 _return_http_data_only=False,  # we want all info, not just data
-                _preload_content=False,
+                _preload_content=True,
                 _request_timeout=None,
                 collection_formats={}
             ))

--- a/kubetest/response.py
+++ b/kubetest/response.py
@@ -1,8 +1,12 @@
 """Response wrapper object for proxied requests to Kubernetes resources."""
 
+import ast
 import json
+import logging
 
 import urllib3
+
+log = logging.getLogger('kubetest')
 
 
 class Response:
@@ -41,20 +45,24 @@ class Response:
         # instances are the HTTP methods on Pods (e.g. http_proxy_get), where
         # `_preload_content=False`. Should that be set to True, it will not return
         # the HTTPResponse but will instead return the response data formatted as
-        # the type specified in the `response_type` param.
-        #
-        # Note that I have found setting _preload_content to True (the default value
-        # for the Kubernetes client) to be difficult to deal with from a generic interface
-        # because it requires you to know the expected output ("str", "dict", ...)
-        # and pass that in as a param, where that could really be abstracted away from
-        # the user. For that reason, it is currently set to False -- this may change
-        # in the future if desired.
+        # the type specified in the `response_type` param. By default, kubetest sets
+        # the _preload_content field to True, so this should generally not be hit.
         if isinstance(self.data, urllib3.HTTPResponse):
             return json.loads(self.data.data)
 
-        # the response data comes back as a string formatted as a python
-        # dictionary might be, where the inner quotes are single quotes
-        # (') instead of the double quotes (") expected by JSON standard.
-        # to remedy this, we just replace single quotes with double quotes.
-        data = self.data.replace("'", '"')
-        return json.loads(data)
+        # The response data comes back as a string. This could be a JSON string,
+        # or something else (text body, error string, etc). Since we've preloaded
+        # the content as a string (see comment above), we can not simply load the
+        # string as JSON as the preloading essentially serializes the Python dict
+        # out to a string, so various values will not parse into JSON (None vs null,
+        # " vs ', etc). To remedy this, we attempt to load the response as an ast
+        # literal - if it fails for any reason, fall back to trying to load JSON.
+        # At the very least the JSON loading will fail with a familiar error which
+        # one would expect from a .json() function.
+        try:
+            data = ast.literal_eval(self.data)
+        except Exception as e:
+            log.debug('failed literal eval of data {} ({})'.format(self.data, e))
+            data = json.loads(self.data)
+
+        return data

